src/OmniSharp.MSBuild/ProjectManager.cs

```diff
@@ -42,6 +43,7 @@ public ProjectToUpdate(string filePath, bool allowAutoRestore)
         private readonly MetadataFileReferenceCache _metadataFileReferenceCache;
         private readonly PackageDependencyChecker _packageDependencyChecker;
         private readonly ProjectFileInfoCollection _projectFiles;
+        private readonly ConcurrentDictionary<string, int> _failedToLoadProjectFiles;
```

Comment:
Is a `ConcurrentDictionary` really needed? MSBuild project loader really doesn't do much in the way of concurrency since it can't use MSBuild concurrently. All of the methods where `_failedToLoadProjectFiles` is updated also update `_projectFiles`, which is just a wrapper around a `List` and a `Dictionary`. In other words, the majority of this code is already synchronous. :smile:

Reply:
I was hoping that this could the first step towards making OmniSharp to be able loading multiple projects in parallel. But for the sake of consistency I will go ahead and change this to Dictionary.

Reply:
FWIW, loading projects in parallel is a big effort. The issue here is that MSBuild handles parallel builds by launching multiple processes to serve as additional nodes. However, doing that cross platform is challenging.